deviceadvisor/tests/utils/datest_utils.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/Types.h>
+#include <iostream>
+#include <stdlib.h>
+
+using namespace Aws::Crt;
+
+namespace DATest_Utils
+{
+
+    const char *ENV_ENDPONT = "DA_ENDPOINT";
+    const char *ENV_CERTI = "DA_CERTI";
+    const char *ENV_KEY = "DA_KEY";
+    const char *ENV_TOPIC = "DA_TOPIC";
+    const char *ENV_THING_NAME = "DA_THING_NAME";
+    const char *ENV_SHADOW_PROPERTY = "DA_SHADOW_PROPERTY";
+    const char *ENV_SHADOW_VALUE_SET = "DA_SHADOW_VALUE_SET";
+    const char *ENV_SHADOW_VALUE_DEFAULT = "DA_SHADOW_VALUE_DEFAULT";
+
+    void s_getenv(const char *env_name, String &var)
```

Comment:
we have an environment api in https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/environment.h

Also you'll want to use the aws-c-* memory management functions rather than raw malloc/free/memset